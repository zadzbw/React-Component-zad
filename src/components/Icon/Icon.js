import React, {PropTypes, Component} from 'react';
import classNames from 'classnames';
import isOneOf from '../../utils/isOneOf';

const _type = {
  size: ['lg', '2x', '3x', '4x', '5x'],
  rotate: [90, 180, 270]
};

export default class Icon extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    size: PropTypes.string,
    fixedWidth: PropTypes.bool,
    spin: PropTypes.bool,
    pulse: PropTypes.bool,
    rotate: PropTypes.number
  };

  render() {
    const {name, size, fixedWidth, spin, pulse, rotate, ...props} = this.props;
    const iconClass = classNames('fa', `fa-${name}`, {
      [`fa-${size}`]: size && isOneOf(_type.size, size),
      [`fa-rotate-${rotate}`]: rotate && isOneOf(_type.rotate, rotate),
      'fa-fw': fixedWidth,
      'fa-spin': spin,
      'fa-pulse': pulse
    });
    return (
      <i {...props} className={iconClass}/>
    );
  }
}
