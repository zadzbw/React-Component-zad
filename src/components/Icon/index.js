/**
 * Created by zad on 16/11/30.
 */
import Icon from './Icon';

export default Icon;