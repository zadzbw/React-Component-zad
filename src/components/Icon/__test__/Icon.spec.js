/**
 * Created by zad on 16/12/7.
 */
import React from 'react';
import {mount} from 'enzyme';
import {mountToJson} from 'enzyme-to-json';
import Icon from '../../Icon';

describe('Icon test', () => {
  it('icon name', () => {
    const wrapper = mount(
      <Icon name={'shield'}/>
    );
    expect(wrapper.find('i').hasClass('fa-shield')).toBeTruthy();
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('size', () => {
    const wrapper = mount(
      <Icon name={'shield'} size={'2x'}/>
    );
    expect(wrapper.find('i').hasClass('fa-shield')).toBeTruthy();
    expect(wrapper.find('i').hasClass('fa-2x')).toBeTruthy();
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('fixedWidth', () => {
    const wrapper = mount(
      <Icon name={'shield'} fixedWidth={true}/>
    );
    expect(wrapper.find('i').hasClass('fa-shield')).toBeTruthy();
    expect(wrapper.find('i').hasClass('fa-fw')).toBeTruthy();
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('spin', () => {
    const wrapper = mount(
      <Icon name={'shield'} spin={true}/>
    );
    expect(wrapper.find('i').hasClass('fa-shield')).toBeTruthy();
    expect(wrapper.find('i').hasClass('fa-spin')).toBeTruthy();
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('pulse', () => {
    const wrapper = mount(
      <Icon name={'shield'} pulse={true}/>
    );
    expect(wrapper.find('i').hasClass('fa-shield')).toBeTruthy();
    expect(wrapper.find('i').hasClass('fa-pulse')).toBeTruthy();
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('rotate', () => {
    const wrapper = mount(
      <Icon name={'shield'} rotate={90}/>
    );
    expect(wrapper.find('i').hasClass('fa-shield')).toBeTruthy();
    expect(wrapper.find('i').hasClass('fa-rotate-90')).toBeTruthy();
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
