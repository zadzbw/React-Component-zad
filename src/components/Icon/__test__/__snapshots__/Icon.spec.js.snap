exports[`Icon test fixedWidth 1`] = `
<Icon
  fixedWidth={true}
  name="shield">
  <i
    className="fa fa-shield fa-fw" />
</Icon>
`;

exports[`Icon test icon name 1`] = `
<Icon
  name="shield">
  <i
    className="fa fa-shield" />
</Icon>
`;

exports[`Icon test pulse 1`] = `
<Icon
  name="shield"
  pulse={true}>
  <i
    className="fa fa-shield fa-pulse" />
</Icon>
`;

exports[`Icon test rotate 1`] = `
<Icon
  name="shield"
  rotate={90}>
  <i
    className="fa fa-shield fa-rotate-90" />
</Icon>
`;

exports[`Icon test size 1`] = `
<Icon
  name="shield"
  size="2x">
  <i
    className="fa fa-shield fa-2x" />
</Icon>
`;

exports[`Icon test spin 1`] = `
<Icon
  name="shield"
  spin={true}>
  <i
    className="fa fa-shield fa-spin" />
</Icon>
`;
